crates/cheatnet/src/cheatcodes/declare.rs

```diff
@@ -39,45 +40,31 @@ impl BlockifierState<'_> {
         let class_hash =
             get_class_hash(contract_artifact.sierra.as_str()).expect("Failed to get class hash");
 
-        // let nonce = blockifier_state
-        //     .get_nonce_at(ContractAddress(patricia_key!(
-        //         TEST_ACCOUNT_CONTRACT_ADDRESS
-        //     )))
-        //     .expect("Failed to get nonce");
+        match blockifier_state.get_compiled_contract_class(&class_hash) {
+            Err(StateError::UndeclaredClassHash(_)) => {
+                // Class is undeclared; declare it.
 
-        // let declare_tx = build_declare_transaction(
-        //     nonce,
-        //     class_hash,
-        //     ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS)),
-        // );
-        // let tx = DeclareTransaction::new(
-        //     starknet_api::transaction::DeclareTransaction::V2(declare_tx),
-        //     // TODO(#358)
-        //     TransactionHash::default(),
-        //     contract_class.clone(),
-        // )
-        // .unwrap_or_else(|err| panic!("Unable to build transaction {err:?}"));
-        // TODO make it correct with compiled class hash
-        blockifier_state
-            .set_contract_class(&class_hash, contract_class)
-            .map_err(EnhancedHintError::from)?;
+                blockifier_state
+                    .set_contract_class(&class_hash, contract_class)
+                    .map_err(EnhancedHintError::from)?;
 
-        // match blockifier_state.get_compiled_contract_class(&class_hash) {
-        //     Err(StateError::UndeclaredClassHash(_)) => {
-        //         // Class is undeclared; declare it.
-
-        //         Ok(class_hash)
-        //         // blockifier_state.set_compiled_class_hash(class_hash, *compiled_class_hash)?;
-        //     }
-        //     Err(error) => Err(error).map_err(TransactionExecutionError::from),
-        //     Ok(_) => {
-        //         // Class is already declared, cannot redeclare
-        //         // (i.e., make sure the leaf is uninitialized).
-        //         Err(TransactionExecutionError::DeclareTransactionError { class_hash })
-        //     }
-        // }
-
-        Ok(class_hash)
+                // NOTE: Compiled class hash is being set to 0 here
+                blockifier_state
+                    .set_compiled_class_hash(class_hash, Default::default())
+                    .unwrap_or_else(|err| panic!("Failed to set compiled class hash: {err:?}"));
+                Ok(class_hash)
```

Comment:
Cannot we compute real compiled class hash from CASM using blockifier's api?

Reply:
Do we need this for anything?

Reply:
Since we communicate that we are consistent with starknet - it would be nice